File: src/api/src/main/java/fr/iut/uca/model/surveys/Question.java
package fr.iut.uca.model.surveys;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Question extends QuestionInfo {

    private String description;

    private final List<String> choices = new ArrayList<>();

    public Question(String title, QuestionType type, String description, List<String> choices) {
        super(title, type);
        this.description = description;
        this.choices.addAll(choices);
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<String> getChoices() {
        return Collections.unmodifiableList(choices);
    }

    @Override
    public boolean equals(Object o) {
        return super.equals(o);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }
}

File: src/app/model/surveys/Question.ts
import {QuestionInfo} from "./QuestionInfo";
import {QuestionType} from "./QuestionType";

export class Question extends QuestionInfo {
  private _description: string
  private _choices: string[]


  constructor(title: string, type: QuestionType, description: string, choices: string[]) {
    super(title, type);
    this._description = description;
    this._choices = choices;
  }


  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get choices(): string[] {
    return this._choices;
  }

  set choices(value: string[]) {
    this._choices = value;
  }
}
